rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -89,17 +90,53 @@ enum MetricValue {
     },
 }
 
+trait Decodable<const SIZE: usize>: Copy {
+    const SIZE: usize = SIZE;
+
+    fn decode_bytes(bytes: [u8; SIZE]) -> Self;
+}
+
+impl Decodable<4> for u32 {
+    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
+        Self::from_le_bytes(bytes)
+    }
+}
+
+impl Decodable<8> for u64 {
+    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
+        Self::from_le_bytes(bytes)
+    }
+}
+
+impl Decodable<8> for f64 {
+    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
+        Self::from_le_bytes(bytes)
+    }
+}
+
 #[derive(Debug, Deserialize)]
 #[serde(tag = "format", rename_all = "lowercase")]
 enum EncodedSeries<T> {
     Array { data: Vec<T> },
+    Base64 { data: String },
 }
 
 impl<T> EncodedSeries<T> {
-    fn into_vec(self) -> Vec<T> {
-        match self {
+    fn into_vec<const SIZE: usize>(self) -> Result<Vec<T>, anyhow::Error>
+    where
+        T: Decodable<SIZE>,
+    {
+        let res = match self {
             EncodedSeries::Array { data } => data,
-        }
+            EncodedSeries::Base64 { data, .. } => BASE64
+                .decode(data.as_bytes())?
+                .chunks_exact(T::SIZE)
```

Comment:
This [ignores trailing data](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks_exact) in the vector, should we raise an error if `data.len()` does not divide `T::SIZE` cleanly?

Reply:
I think we're working on transforming the TryFromSliceError that we would get into something more generic, to match the anyhow::Error return type.

I'm curious though -- what's the exact reason we would need this kind of error handling for trailing data? Wouldn't we be expecting sequences of exact u32 or f64 anyways?

Reply:
Added the length check, decided to not check using `.chunks` and to just check the length of the decoded bytes with an `if` in the end. Added a test as well.